.travis.yml

```diff
@@ -107,17 +108,55 @@ before_script:
     chmod +x ./cc-test-reporter
     ./cc-test-reporter before-build
   fi
+- export -f travis_fold
+- export -f travis_time_start
+- export -f travis_time_finish
 
 script:
-- if [[ "$PHPLINT" == "1" ]]; then find -L . -path ./vendor -prune -o -path ./node_modules -prune -o -name '*.php' -print0 | xargs -0 -n 1 -P 4 php -l; fi
-- if [[ "$PHPCS" == "1" ]]; then vendor/bin/phpcs -v --runtime-set ignore_warnings_on_exit 1; fi
-- if [[ "$CHECKJS" == "1" ]]; then grunt check:js; fi
-- if [[ "$COVERAGE" != "1" ]] && [[ ${TRAVIS_PHP_VERSION:0:1} == "5" || $TRAVIS_PHP_VERSION == hhv* ]]; then phpunit -c phpunit.xml; fi
-- if [[ "$COVERAGE" != "1" ]] && [[ ${TRAVIS_PHP_VERSION:0:1} == "7" || $TRAVIS_PHP_VERSION == "nightly" ]]; then phpunit -c phpunit.xml; fi
+# JavaScript checks
+- |
+  if [[ "$CHECKJS" == "1" ]]; then
+    travis_fold start "JavaScript.check" && travis_time_start
+    grunt check:js
+    travis_time_finish && travis_fold end "JavaScript.check"
+  fi
+# JavaScript tests
+- |
+  if [[ "$CHECKJS" == "1" ]]; then
+    travis_fold start "JavaScript.tests" && travis_time_start
+    yarn test
+    travis_time_finish && travis_fold end "JavaScript.tests"
+  fi
+# PHP Linting
+- |
+  if [[ "$PHPLINT" == "1" ]]; then
```

Comment:
Folding is not needed here - if you change the command slightly, the output is a one-liner already (except when parse errors are encountered, but that's the point):

```bash
- if find -L . -path ./vendor -prune -o -path ./node_modules -prune -o -name '*.php' -exec php -l {} \; | grep "^[Parse error|Fatal error]"; then exit 1; fi
```

To see it in action: https://travis-ci.org/WordPress-Coding-Standards/WordPress-Coding-Standards/jobs/339562207#L537

Reply:
I do prefer to have to entire list of files being linted available to be able to double check things.